### PR TITLE
Improve the DSL docs page

### PR DESCRIPTION
**Pull request by KipHamiltons**
Mostly pretty easy stuff. Opted for `rst` features instead of HTML as well.

### CODE REVIEW

File: docs/dsl.rst [hunk 1]
```diff
@@ -1,4 +1,4 @@
-include .special.rst
+.. include:: .special.rst
```
Comment (KipHamiltons) [inline docs/dsl.rst:1]: This is literally at the top of the page. Yikes
https://nuclear.readthedocs.io/en/latest/dsl.html

File: docs/dsl.rst [hunk 2]
```diff
@@ -10,18 +10,14 @@ On Statements
...
 The on statement can be seen as containing three main parts.  The DSL Request, the Runtime Arguments, and the Callback.
 
-.. raw:: html
-
-    <font color="blue">On<...></font><font color="red">(Runtime, ... )</font><font color="green">.then(function);</font>
-    <br /><br />
-    <font color="blue">DSL Request</font><br />
-
-This is :red:`red !` And :blue:`this part is blue`.
+:blue:`On<Dsl, Word(s)>`\(:red:`Runtime, Args`).then(:green:`CallbackFunction`);
```
Comment (KipHamiltons) [inline docs/dsl.rst:17]: If we want to use `rst`, we should actually use it.

Also, that escaped paren is deliberate. It's required after the backtick

File: docs/dsl.rst [hunk 3]
```diff
@@ -30,22 +26,18 @@ For reactions to occur, at least one Binding DSL word should be present in the D
...
 
-.. raw:: html
-
-    <font color="green">Callback</font>
-
+:green:`Callback`
+-----------------
 Finally, the developer can define the callback which will execute when the reaction is triggered during runtime.  The
-callback can be defined using a C++ lambda function.
+callback can be defined using a C++ lambda expression.
```
Comment (KipHamiltons) [inline docs/dsl.rst:40]: ACKSHUALLY 🤓

File: docs/dsl.rst [hunk 4]
```diff
@@ -195,14 +187,9 @@ Note that data can be emitted under varying scopes:
 Local Emitting
 --------------
 
-These emissions send data to the local instance of the NUClear powerplant.  There are a number of scopes under which
+These emissions send data to the local instance of the NUClear PowerPlant.  There are a number of scopes under which
 these emissions can take place:
 
-.. todo::
-
-    Trent - I need to decide and get consistent on what we will call the powerPlant.  Should it be PowerPlant or
```
Comment (KipHamiltons) [inline docs/dsl.rst:203]: Making an executive decision here

**Review by TrentHouliston — APPROVED** [review]

**Pull request merged**